2025-06-18 10:43:22,250 - ERROR - Traceback (most recent call last):
  File "D:\CS\visual perception\BadmintonAnalysis\main.py", line 220, in <module>
    ball_detect(res_video_path, f"{result_path}")
  File "D:\CS\visual perception\BadmintonAnalysis\src\tools\BallDetect.py", line 126, in ball_detect
    results = shuttle_tracker.detect_shuttle(imgs_torch[0].permute(1, 2, 0).cpu().numpy())
  File "D:\anaconda3\envs\visualPerception\lib\site-packages\ultralytics\engine\model.py", line 1160, in __getattr__
    return self._modules["model"] if name == "model" else getattr(self.model, name)
  File "D:\anaconda3\envs\visualPerception\lib\site-packages\torch\nn\modules\module.py", line 1614, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'DetectionModel' object has no attribute 'detect_shuttle'

2025-06-18 11:47:23,866 - ERROR - Traceback (most recent call last):
  File "D:\CS\visual perception\BadmintonAnalysis\main.py", line 220, in <module>
    ball_detect(res_video_path, f"{result_path}")
  File "D:\CS\visual perception\BadmintonAnalysis\src\tools\BallDetect.py", line 138, in ball_detect
    results[f"{count + start_frame}"] = {"visible": 1, "x": cx, "y": cy}
UnboundLocalError: local variable 'cx' referenced before assignment

